src/nomad_simulations/model_system.py

```diff
@@ -317,59 +562,79 @@ class AtomicCell(Cell):
-        if self.positions is not None:
-            if len(self.positions) != len(self.atoms_state):
-                logger.error(
-                    'Length of `AtomicCell.positions` does not coincide with the length of the `AtomicCell.atoms_state`.'
-                )
-                return None
-            ase_atoms.set_positions(self.positions.to('angstrom').magnitude)
-        else:
-            logger.error('Could not find `AtomicCell.positions`.')
-            return None
-
-        # Lattice vectors
-        if self.lattice_vectors is not None:
-            ase_atoms.set_cell(self.lattice_vectors.to('angstrom').magnitude)
-        else:
-            logger.info('Could not find `AtomicCell.lattice_vectors`.')
+        self.ase_atoms = ase.Atoms(
+            symbols=[atom_state.chemical_symbol for atom_state in self.atoms_state],
+            pbc=self.periodic_boundary_conditions,
+            positions=self.positions.to('angstrom').magnitude,
+            cell=self.lattice_vectors.to('angstrom').magnitude,
+        )
+        return self
 
-        return ase_atoms
+    def __init__(self, m_def: Section = None, m_context: Context = None, **kwargs):
+        super().__init__(m_def, m_context, **kwargs)
+        self.name = self.m_def.name
+        self.analyze_geometry = kwargs.get('analyze_geometry', False)
+        self.ase_atoms = None
 
-    def normalize(self, archive, logger) -> None:
+    def normalize(self, archive, logger: BoundLogger):
+        """When `analyze_geometry` is `true`, perform the geometry analysis.
+        All intermediate artifacts are stored in the `AtomicCell` section.
+        Only `GeometryDistribution` sections are retained in the archive written to disk."""
         super().normalize(archive, logger)
+        if self.analyze_geometry:
+            # set up neighbor list
+            if not self.geometry_analysis_cutoffs:
+                self.geometry_analysis_cutoffs = ase_nl.natural_cutoffs(
+                    self.ase_atoms, mult=3.0
+                )
+            self.neighbor_list = ase_nl.build_neighbor_list(
+                self.ase_atoms,
+                self.geometry_analysis_cutoffs,
+                self_interaction=False,
+            )
 
-        # Set the name of the section
-        self.name = self.m_def.name if self.name is None else self.name
+            # set up distributions
+            self.simple_distributions = DistributionFactory(
+                self.ase_atoms,
+                self.neighbor_list,
+            ).produce_distributions()
+
+            self.histogram_distributions, self.distributions = [], []
+            for distribution in self.simple_distributions:
+                if distribution.type == 'distances':
+                    bins = np.arange(
+                        np.arange(0, max(self.geometry_analysis_cutoffs), 0.001)
+                    )
+                elif distribution.type == 'angles':
+                    bins = np.arange(0, 180, 0.01)
```

Comment:
> I would still apply the same parameters as for distances for generality of usage.

I chose the default based on the order of accuracy limit. Since angle and distance histograms shouldn't be compared, comparing bin size is also non-sensical.

Reply:
I was not saying that angles and distances should literally have the same bin spacing magnitude, and I did not mean that they should not have default values. My point here was that The GeometryDistribution() should contain an attribute called `bin_spacing`, such that default values can be set individually within the classes DihedralGeometryDistribution(), etc. That way they can also be changed from the default if necessary.

Reply:
> My point here was that The GeometryDistribution() should contain an attribute called bin_spacing

Yes, I understood this from your previous comment too. Still, I'd advise against it for the following 2 reasons:

1. We want to make these histograms searchable. At NOMAD's scale, this means constructing an index. Schema uniformity plays a pretty big role (will defer to Lauri or Markus here) in achieving that performance. If storage was more of a concern, I'd instead go with a sparse representation. Actually, now that distances and angles are dealt with separately, we could even remove the `bins` altogether.
2. Schema complexity: I understand the desire to placate to each use case, but this comes with trade-offs (such as point 1). This setup is generic enough that we can accomodate all covalent distributions and a bit beyond. These classes could be used as a jumping-off point for similar distribution analyses, but at different scales. That's fine, but I'd store that under a different name (and different issue).

Reply:
Regarding the sparse vs fixed binning, I'm actually not sure which would be better. I could see to deploy a ES instance and try it out myself.

Reply:
I tried out the performance a bit this week: results are unclear. Both seem viable. I'll defer this to a follow-up issue.